scality_sproxyd_client/sproxyd_client.py

```diff
@@ -142,6 +142,19 @@ def __repr__(self):
         return ret % (self.sproxyd_urls_set, self.conn_timeout,
                       self.proxy_timeout)
 
+    def get_url_for_object(self, name):
+        '''Get an absolute URL from which the object `name` can be accessed.
+
+        Only healthy Sproxyd endpoints are concidered to form the base object
+        path. The object `name` is properly escaped by `urllib.quote` which
+        preserves '/' by default.
+        '''
+        try:
+            sproxyd_url = self.sproxyd_urls.next()
+        except StopIteration:
+            raise exceptions.SproxydException("No Sproxyd endpoint alive !")
```

Comment:
I guess I wanted to add some drame.

I should have gone with "OOhhh nnnooessss, something went wrong !!!!!!!!!!1"